src/core/public/jest.config.js

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+module.exports = {
+  preset: '@kbn/test',
+  rootDir: '../../..',
+  roots: ['<rootDir>/src/core'],
+  testRunner: 'jasmine2',
```

Comment:
Is this really necessary? I'd like to make sure this is needed in these new configs before copying it forward.

Reply:
I duplicated this file to overcome a limitation on the "check_jest_configs" logic, which uses a list of `root` folders that is used to determine both the roots of both integration tests' config files and unit tests' config files.

Once a test file is matched against a root, a jest configuration file of the corresponding type (integration | unit) is expected at the root folder.

I suppose we could improve the logic in order to match multiple roots and search for existing config files in multiple places instead. I'll give it a try.

Reply:
I'm referring to the `testRunner` config, I think it's good to have separate config files.

Reply:
Oh I see.

According to the docs, the default test runner on jest-26.x is already `jasmine2`, but on 27.x they have [switched the default](https://jestjs.io/blog/2021/05/25/jest-27#flipping-defaults) to `jest-circus` (which should be pretty much compatible out of the box).

Thus, we can remove this configuration as it is unnecessary for now (we are using ^26.6.3). Then, we can create separate PRs to try and remove the `testRunner` configuration everywhere and to upgrade to jest-27.x.